docs/articles/documentation/test-api/handling-native-dialogs.md

```diff
@@ -0,0 +1,137 @@
+---
+layout: docs
+title: Handling Native Dialogs
+permalink: /documentation/test-api/handling-native-dialogs.html
+---
+# Handling Native Dialogs
+
+TestCafe allows you to handle native browser dialogs whenever they are invoked during test run.
+You can close alert and beforeunload dialogs, choose either option in confirm dialogs or provide text to type into prompt dialogs.
+
+You can also access the history of the invoked native dialogs so that you can assert that a certain dialog appeared or not.
+
+This topic contains the following sections:
+
+* [Dialog Handler](#dialog-handler)
+* [Dialog History](#dialog-history)
+
+## Dialog Handler
+
+To handle native dialogs invoked during the test run, specify a handler function
+by using the `setNativeDialogHandler` method of the
+[test controller](test-code-structure.md#test-controller).
+
+```text
+setNativeDialogHandler(function(type, text), options)
+```
+
+Parameter  | Type                           | Description
+---------- | ------------------------------ | -------------
+`function` | Function &#124; ClientFunction | A regular or [client function](executing-code-in-the-browser/client-functions.md) that will be triggered whenever a native dialog is invoked.
+`options`  | Object                         | See [Client Function Options](executing-code-in-the-browser/client-functions.md#options).
+
+The `function` has the following arguments.
+
+Argument | Type   | Description
+-------- | ------ | -------------
+`type`   | String | The type of the native dialog. `'alert' | 'confirm' | 'beforeunload' | 'prompt'`.
```

Comment:
``'alert' | 'confirm' | 'beforeunload' | 'prompt'`` 
This part is not rendered correctly, I guess because it's malformed markdown